homework/index.js

```diff
@@ -27,21 +31,145 @@
         elem.setAttribute(key, value);
       }
     });
+
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
-      }
+  function addRow(parent, labelText, rowValue) {
```

Comment:
I see you have implemented the `addRow` function I asked for. You asked for a hint and I was planning to demo it during the lecture. Good that you picked it up from the lecture.